daemons/ipa-slapi-plugins/ipa-pwd-extop/prepost.c

```diff
@@ -68,6 +68,12 @@
 #define IPAPWD_OP_ADD 1
 #define IPAPWD_OP_MOD 2
 
+/* Partial DN of the container for app passwords */
+#define APPPW_CONTAINER_DN "cn=apps,cn=accounts"
+
+/* Maximum number of allowed app passwords */
+#define APPPW_MAX_COUNT 100
```

Comment:
Is 100 arbitrary or only a temporary setting to get the basic functionality working?

Reply:
Yes, in practice. I did not find other way to implement this, as I did not find any function of the slapi plugin to search entries in the LDAP database - therefore, I am not able to easily know what `apppw` objects are already stored in the LDAP database. I picked the number 100, as it is certainly sufficient quantity of app passwords for a single user, while it can be easily enforced by a regex, and it seemed to me as reasonably limiting the performance impact of the new code - but this is where I may be wrong.

Reply:
To do internal search for entries, use a new `Slapi_PBlock` and do `slapi_search_internal_set_pb()` / `slapi_search_internal_pb()` combination. See, for example, `ipa_cldap_get_domain_entry()` implementation in `daemon/ipa-slapi-plugins/ipa-cldap/ipa_cldap_netlogon.c`